app/assets/javascripts/discourse/app/components/number-field.js

```diff
@@ -5,6 +5,22 @@ import discourseComputed from "discourse-common/utils/decorators";
 export default TextField.extend({
   classNameBindings: ["invalid"],
 
+  keyDown: function (e) {
+    const key = e.which;
+
+    return (
+      key === 13 || // Enter
+      key === 8 || // Backspace
+      key === 9 || // Tab
+      key === 46 || // Delete
+      ((key === 109 || key === 189) && // Negative sign
+        parseInt(this.get("min"), 10) < 0) ||
+      (key >= 35 && key <= 40) || // Cursor keys
+      (key >= 48 && key <= 57) || // Numbers
+      (key >= 96 && key <= 105) // Numpad
+    );
```

Comment:
I think we could use `key` instead of `which` which would allow us to match on the string and remove the need for these comments.

Also note that `which` is deprecated: https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/which

Reply:
Excellent suggestion! 🙂